src/request.ts

```diff
@@ -152,6 +165,97 @@ export async function getRequestLinkDetails({
 	return x
 }
 
+export async function prepareXchainRequestLinkFulfillmentTransaction({
+	senderAddress,
+	recipientAddress,
+	destinationChainId,
+	destinationToken,
+	fromAmount,
+	link,
+	squidRouterUrl,
+	provider,
+	apiUrl = 'https://api.peanut.to/',
+}: IPrepareXchainRequestLinkFulfillmentTransactionProps): Promise<interfaces.IPrepareDepositTxsResponse> {
```

Comment:
thought: maybe we should rename `IPrepareDepositTxsResponse` to something else? since it's now more generic and used in two places